src/dynamixel/servos/servo.hpp

```diff
@@ -0,0 +1,243 @@
+#ifndef DYNAMIXEL_SERVOS_SERVO_HPP_
+#define DYNAMIXEL_SERVOS_SERVO_HPP_
+
+#include <cassert>
+#include <cmath>
+#include <stdint.h>
+
+#include "../instruction_packet.hpp"
+#include "../status_packet.hpp"
+#include "base_servo.hpp"
+#include "../instructions/ping.hpp"
+#include "../instructions/read.hpp"
+#include "../instructions/write.hpp"
+#include "../instructions/reg_write.hpp"
+#include "../instructions/action.hpp"
+#include "../instructions/factory_reset.hpp"
+#include "../instructions/sync_write.hpp"
+#include "model_traits.hpp"
+#include "../errors/error.hpp"
+
+#define MODEL_NAME(Name) \
+    std::string model_name() const override \
+    { \
+        return #Name; \
+    }
+
+#define READ_FIELD(Name) \
+    static InstructionPacket<typename Servo<Model>::protocol_t> get_##Name(typename Servo<Model>::protocol_t::id_t id) \
+    { \
+        return typename Servo<Model>::read_t(id, Servo<Model>::ct_t::Name, sizeof(typename Servo<Model>::ct_t::Name##_t)); \
+    } \
+    \
+    static typename Servo<Model>::ct_t::Name##_t  parse_##Name(typename Servo<Model>::protocol_t::id_t id, const StatusPacket<typename Servo<Model>::protocol_t>& st) \
+    { \
+        typename Servo<Model>::ct_t::Name##_t res; \
+        Servo<Model>::protocol_t::unpack_data(st.parameters(), res); \
+        return res; \
+    } \
+    \
+    InstructionPacket<typename Servo<Model>::protocol_t> get_##Name() const override \
+    { \
+        return Model::get_##Name(this->_id); \
+    } \
+    \
+    long long int parse_##Name(const StatusPacket<typename Servo<Model>::protocol_t>& st) const override \
+    { \
+        return static_cast<long long int>(Model::parse_##Name(this->_id, st)); \
+    }
+
+#define WRITE_FIELD(Name) \
+    static inline InstructionPacket<typename Servo<Model>::protocol_t> set_##Name(typename Servo<Model>::protocol_t::id_t id, typename Servo<Model>::ct_t::Name##_t value) \
+    { \
+            return typename Servo<Model>::write_t(id, Servo<Model>::ct_t::Name, Servo<Model>::protocol_t::pack_data(value)); \
+    } \
+    \
+    static inline InstructionPacket<typename Servo<Model>::protocol_t> reg_##Name(typename Servo<Model>::protocol_t::id_t id, typename Servo<Model>::ct_t::Name##_t value) \
+    { \
+            return typename Servo<Model>::reg_write_t(id, Servo<Model>::ct_t::Name, Servo<Model>::protocol_t::pack_data(value)); \
+    } \
+    \
+    inline InstructionPacket<typename Servo<Model>::protocol_t> set_##Name(typename Servo<Model>::ct_t::Name##_t value)  const \
+    { \
+            return Model::set_##Name(this->_id, value); \
```

Comment:
I'm not sure that I understand your question.
We always have 2 versions of a method:
The static version, accesed trough the model, e.g. `Mx28::set_moving_speed(id, speed)`,
and the instance version, which calls the static version of the same function, but passing the stored id.